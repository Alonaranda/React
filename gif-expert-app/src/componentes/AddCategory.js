import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setValue] = useState('');

    const handleInputText = (e) => {
        setValue(e.target.value);
        //console.log('Change', inputValue);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        //console.log('Submit!', inputValue);
        if(inputValue.trim().length > 2){
            setCategories(cate => [inputValue, ...cate, ]);
            setValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputText}
                placeholder='Buscado tu Gif'
            ></input>
        </form>
        
    )
}

//PropTypes
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
