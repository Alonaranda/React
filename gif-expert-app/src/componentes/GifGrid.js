import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types';
//import { getGifs } from "../helpers/getGifs";


export const GifGrid = ({ category }) => {
  //Al ejecutar una funcion dentro de una api, se vuelve un ciclo infinito
  //UseEffect hace que no hagamos ciclo infinito
  //Ejecuta cierto codigo de manera condicional
  //El arreglo vacio, indica que solo se ejecuta una vez

  
    
  //const [images, setImages] = useState([]);
    //Usamos helpers
    const {data:images, loading} = useFetchGifs(category);
    //console.log(data, loading);

    /*
    useEffect(() => {
        getGifs(category)
            .then(imgs =>{
                setImages(imgs);
        })
    }, [category]);//Si la category cambia, se vuelve a ejecutar
    */
  

  return (
    <>
      <h3 className="animate_animated animate_fadeIn">{category}</h3>
      {loading && <p className="animate_animated animate_flash">Loading...</p>}

      <div className="card-grid">
        {images.map(img => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
