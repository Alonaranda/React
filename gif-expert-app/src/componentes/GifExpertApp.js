import React, {useState} from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = ({defaultCategories=[]}) => {
    //const categories = ['Inuyasha', 'Naruto', 'Samurai X'];
    const [categories, setCategories] = useState(defaultCategories);
    /*
    const handleAdd = () => {
        //setCategories([...categories, 'DeathNote']);
        setCategories( cate => [...cate, 'DeathNote']);
    }*/

    //Para pasar datos en el <AddCategory/> agregamos que vamos a utilizar
    return (
        <>
            <h2>GifExpertApp</h2>
             
            <AddCategory setCategories={setCategories}/>
            <hr/>
            <ol>
                {
                    categories.map(category => 
                        <GifGrid 
                        key={category}
                        category={category}
                        />
                    //<li key={category}>{category}</li>
                    ) //El key no debe ser el indice, casi siempre es el id de la BD
                }
            </ol>
        </>
    )
}

