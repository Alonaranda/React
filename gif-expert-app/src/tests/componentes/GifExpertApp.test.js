import React from 'react'
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../../componentes/GifExpertApp';


describe('Probando <GifExpertApp/>', () => {
    test('Snapshot', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['Inuyasha', 'Naruto'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});