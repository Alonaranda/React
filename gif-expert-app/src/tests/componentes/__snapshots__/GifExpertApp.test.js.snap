// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <GifExpertApp/> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Inuyasha"
      key="Inuyasha"
    />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;

exports[`Probando <GifExpertApp/> Snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
